packages/grafana-ui/src/components/OptionsUI/slider.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useCallback } from 'react';
+import { FieldConfigEditorProps, SliderFieldConfigSettings } from '@grafana/data';
+import { Slider } from '../Slider/Slider';
+
+export const SliderValueEditor: React.FC<FieldConfigEditorProps<number | number[], SliderFieldConfigSettings>> = ({
```

Comment:
I'm not convinced about the typing here.

I assume the `number[]` typing is to enable range sliders. But I would opt for range slider (`addRangeSlider`) to be a separate method on both `FieldConfigEditorBuilder` and  `PanelOptionsEditorBuilder`. 

The SlideValueEditor is basically a specific UI implementation of numeric option hence it should have pretty much the same interface and implementation as [numeric option](https://github.com/grafana/grafana/blob/62aa15f76074d028599feb0c628cf13f533c6245/packages/grafana-ui/src/components/OptionsUI/number.tsx#L10-L9):

```
  const slider: StandardEditorsRegistryItem<number> = {
    id: 'slider',
    name: 'Slider', 
    description: 'Allows numeric values input',
    editor: SliderValueEditor as any,
  };
```

And then the range slider:

```
  const rangeSlider: StandardEditorsRegistryItem<[number, number]> = {
    id: 'rangeSlider',
    name: 'Range slider', 
    description: 'Allows numeric range input',
    editor: RangeSliderValueEditor as any,
  };
```

The distinction between those two will simplify the logic behind the initial value.

Reply:
Aha, I agree on making two different components out of this to make both of them simpler and easier to understand.

Reply:
I agree with this as well, the Variable system has shown that having string | string[] as a type introduces a lot of confusion and bugs, great suggestion @dprokop 🌟

Reply:
While working on this I had the same thoughts, but didn't want to opt out the range feature. But I als o think it's better to reiterate once the initial slider component is implemented.